readthedocs/oauth/services/base.py

```diff
@@ -235,11 +235,13 @@ def sync(self):
             r.remote_id for r in all_remote_repositories if r is not None
         ]
         (
-            self.user.remote_repository_relations.exclude(
-                remote_repository__remote_id__in=repository_remote_ids,
+            self.user.remote_repository_relations.filter(
+                account=self.account,
                 remote_repository__vcs_provider=self.vcs_provider_slug,
```

Comment:
I don't follow this change.

Previously, we were using `remote_repository__vcs_provider` under `.exclude`, but now we are using the same `remote_repository__vcs_provider` under `.filter`.

Can you expand the context behind this change?

Reply:
We want to do the deletions over that specific type of provider, previously we were filtering that type of provider only when excluding, possibly deleting remote repositories that weren't part of that provider.